mysql-test/suite/rocksdb/t/use_direct_reads_writes.test

```diff
@@ -5,43 +5,37 @@
 # caused an assertion in RocksDB.  Now it should not be allowed and the
 # server will not start with that configuration
 
-# Write file to make mysql-test-run.pl expect the "crash", but don't restart
-# the server until it is told to
---let $_server_id= `SELECT @@server_id`
---let $_expect_file_name= $MYSQLTEST_VARDIR/tmp/mysqld.$_server_id.expect
---exec echo "wait" >$_expect_file_name
-shutdown_server 10;
+--let SEARCH_FILE=$MYSQLTEST_VARDIR/tmp/rocksdb.use_direct_reads_writes.err
 
-# Clear the log
---exec echo "" >$MYSQLTEST_VARDIR/log/mysqld.1.err
+# Expect server crash on startup with invalid combination of arguments
+--source include/expect_crash.inc
+shutdown_server 10;
 
 # Attempt to restart the server with invalid options
---exec echo "restart:--rocksdb_use_direct_reads=1 --rocksdb_allow_mmap_reads=1" >$_expect_file_name
---sleep 0.1  # Wait 100ms - that is how long the sleep is in check_expected_crash_and_restart
---exec echo "restart:" >$_expect_file_name
-
-# Cleanup
---enable_reconnect
---source include/wait_until_connected_again.inc
---disable_reconnect
+--let $restart_parameters=restart:--log-error=$SEARCH_FILE --rocksdb_use_direct_reads=1 --rocksdb_allow_mmap_reads=1
+--replace_result $SEARCH_FILE LOG_FILE
+--source include/start_mysqld.inc
```

Comment:
What about defaults-file=...my.cnf contents? What does machine ps show during the hung launch? Does MTR --debug or --verbose (both can be repeated several times IIRC) show what it's actually trying to invoke or what it is waiting for? We can take this offline from PR review if more convenient